addons/hw_escpos/escpos/escpos.py

```diff
@@ -885,14 +885,23 @@ def cut(self, mode=''):
             self._raw(PAPER_FULL_CUT)
 
 
-    def cashdraw(self, pin):
-        """ Send pulse to kick the cash drawer """
-        if pin == 2:
-            self._raw(CD_KICK_2)
-        elif pin == 5:
-            self._raw(CD_KICK_5)
-        else:
-            raise CashDrawerError()
+    def cashdraw(self, pin, tries=5):
+        """ Send pulse to kick the cash drawer
+
+        With some printers the drawer will not open after one pulse, for this reason we will check
+        the drawer status up to 'tries' times and send a new pulse if the drawer is still closed. If the
+        drawer status is open, we will stop sending pulses.
+        """
+        for i in range(tries):
+            if pin == 2:
+                self._raw(CD_KICK_2)
+            elif pin == 5:
+                self._raw(CD_KICK_5)
+            else:
+                raise CashDrawerError()
+
+            if i != tries - 1 and not self.get_printer_status()['printer']['drawer_pin_high']:
```

Comment:
wa can break even if i == tries-1 it will work

Reply:
No, it works like this. The second check should not be done if I == tries - 1 because it will break eather way then.

Reply:
So put the 'if not self.get_printer_status()['printer']['drawer_pin_high']'  before it will simpler and no need of the tries-1

Reply:
That was the first version, but you didn't like it for if for some reason the detection switch would not work the drawer would not get a pulse at all,what was a valid remark.